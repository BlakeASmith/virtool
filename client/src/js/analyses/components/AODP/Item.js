import React, { useCallback } from "react";
import { connect } from "react-redux";

import { setActiveHitId } from "../../actions";
import { getActiveHit, getMatches } from "../../selectors";
import { AnalysisViewerItem } from "../Viewer/Item";

export const AODPItem = ({ active, name, id, style, onSetActiveId }) => {
    const handleClick = useCallback(() => onSetActiveId(id), [id]);

    return (
        <AnalysisViewerItem onClick={handleClick} style={style} selected={active}>
            <strong>{name}</strong>
        </AnalysisViewerItem>
    );
};

const mapStateToProps = (state, ownProps) => {
    const activeId = getActiveHit(state).id;
    const matches = getMatches(state);
    const match = matches[ownProps.index];

    const { id, name } = match;
    return { id, name, active: activeId === id };
};

const mapDispatchToProps = dispatch => ({
    onSetActiveId: id => {
        dispatch(setActiveHitId(id));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(AODPItem);
