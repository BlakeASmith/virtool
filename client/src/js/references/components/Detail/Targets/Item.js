import React from "react";
import styled from "styled-components";
import { BoxGroupSection, Icon, Flex, FlexItem } from "../../../../base";

const NameIcon = styled.div`
    display: flex;
    justify-content: space-between;
`;

export const TargetItem = ({ name, description, onEdit, onRemove }) => {
    return (
        <BoxGroupSection>
            <NameIcon>
                {name}
                <FlexItem grow={1} shrink={1}>
                    <Flex alignItems="center" className="pull-right">
                        <Icon name="edit" bsStyle="warning" tip="Modify" onClick={onEdit} />
                        <FlexItem pad>
                            <Icon name="trash" bsStyle="danger" tip="Remove" onClick={onRemove} />
                        </FlexItem>
                    </Flex>
                </FlexItem>
            </NameIcon>
            {description}
        </BoxGroupSection>
    );
};
