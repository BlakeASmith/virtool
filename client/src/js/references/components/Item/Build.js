import React from "react";
import { Link } from "react-router-dom";
import { RelativeTime } from "../../../base";

export const ReferenceItemBuild = ({ id, latestBuild, progress }) => {
    if (latestBuild && progress === 100) {
        return (
            <tr>
                <th>Latest Build</th>
                <td>
                    <Link to={`/refs/${id}/indexes/${latestBuild.id}`}>Index {latestBuild.version}</Link>
                    <div className="text-muted" style={{ fontSize: "12px" }}>
                        Created <RelativeTime time={latestBuild.created_at} /> by {latestBuild.user.id}
                    </div>
                </td>
            </tr>
        );
    }

    return (
        <tr>
            <th>Latest Build</th>
            <td>No Build Found</td>
        </tr>
    );
};
