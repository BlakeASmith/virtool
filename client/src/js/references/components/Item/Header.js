import PropTypes from "prop-types";
import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import { Icon, Panel, RelativeTime } from "../../../base";

const ReferenceItemHeaderLink = styled(Link)`
    font-weight: bold;
`;

const ReferenceItemHeaderTitle = styled.div`
    align-items: center;
    display: flex;
    justify-content: space-between;
`;

export const ReferenceItemHeader = ({ createdAt, id, name, userId }) => (
    <Panel.Heading>
        <ReferenceItemHeaderTitle>
            <ReferenceItemHeaderLink to={`/refs/${id}`}>{name}</ReferenceItemHeaderLink>
            <Link to={{ state: { newReference: true, cloneReference: true, id } }}>
                <Icon name="clone" tip="Clone" />
            </Link>
        </ReferenceItemHeaderTitle>
        <small>
            Created <RelativeTime time={createdAt} /> by {userId}
        </small>
    </Panel.Heading>
);

ReferenceItemHeader.propTypes = {
    createdAt: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    userId: PropTypes.string.isRequired
};
