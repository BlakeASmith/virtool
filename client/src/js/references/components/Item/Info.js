import styled from "styled-components";
import { getFontSize, getFontWeight } from "../../../app/theme";
import { Box } from "../../../base";

export const ReferenceItemInfo = styled(Box)`
    background-color: #f7fafc;
    border-color: #718096;
    border-radius: 3px;
    font-weight: bold;
    height: 100%;
    margin: 0;
    padding: 8px 11px;

    h4 {
        font-weight: ${getFontWeight("thick")};
        font-size: ${getFontSize("md")};
        margin: 0 0 3px;
    }

    p {
        font-size: ${getFontSize("sm")};
        font-weight: ${props => props.theme.fontWeight.normal};
        margin: 0;
    }
`;
