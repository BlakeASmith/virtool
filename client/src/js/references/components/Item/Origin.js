import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { ExternalLink } from "../../../base";

export const ReferenceItemOrigin = ({ clonedFrom, importedFrom, remotesFrom }) => {
    if (clonedFrom) {
        return (
            <tr>
                <th>Cloned From</th>
                <td>
                    <Link to={`/refs/${clonedFrom.id}`}>{clonedFrom.name}</Link>
                </td>
            </tr>
        );
    }

    if (importedFrom) {
        return (
            <tr>
                <th>Imported From</th>
                <td>{importedFrom.name}</td>
            </tr>
        );
    }

    if (remotesFrom) {
        return (
            <tr>
                <th>Remotes from</th>
                <td>
                    <ExternalLink href={`https://www.github.com/${remotesFrom.slug}`}>{remotesFrom.slug}</ExternalLink>
                </td>
            </tr>
        );
    }

    return (
        <tr>
            <th>Created</th>
            <td>No File</td>
        </tr>
    );
};

ReferenceItemOrigin.propTypes = {
    clonedFrom: PropTypes.object,
    importedFrom: PropTypes.object,
    remotesFrom: PropTypes.object
};
