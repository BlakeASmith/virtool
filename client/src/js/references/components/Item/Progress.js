import React from "react";
import styled from "styled-components";
import { ProgressBar } from "../../../base";

const ReferenceItemProgressPlaceholder = styled.div`
    height: 4px;
    width: 100%;
`;

export const ReferenceItemProgress = ({ now }) => {
    if (now === 100) {
        return <ReferenceItemProgressPlaceholder />;
    }

    return <ProgressBar bsStyle="warning" now={now} affixed />;
};
