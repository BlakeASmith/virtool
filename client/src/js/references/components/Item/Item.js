import React from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import { BoxGroup, ProgressBar, Table } from "../../../base";
import { getReferenceItemProgress } from "../../selectors";
import { ReferenceItemHeader } from "./Header";
import { ReferenceItemOrigin } from "./Origin";
import { ReferenceItemBuild } from "./Build";

const ReferenceItemTable = styled(Table)`
    tr {
        td,
        th {
            min-width: 70px;
        }

        &:first-child {
            border-top: 1px solid #dddddd;
        }
    }
`;

const StyledReferenceItem = styled(BoxGroup)`
    box-shadow: 1px 1px 2px 0 #d5d5d5;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    .progress {
        flex: 0 0 auto;
        width: 100%;
        height: 5px;
    }

    table {
        margin-bottom: 0;
    }

    th {
        padding-left: 15px !important;
    }
`;

export const ReferenceItem = ({
    clonedFrom,
    createdAt,
    id,
    importedFrom,
    latestBuild,
    name,
    organism,
    progress,
    remotesFrom,
    userId
}) => {
    return (
        <StyledReferenceItem>
            <ReferenceItemHeader id={id} createdAt={createdAt} name={name} userId={userId} />
            <ReferenceItemTable>
                <tbody>
                    <tr>
                        <th>Organism</th>
                        <td className="text-capitalize">{organism || "unknown"}</td>
                    </tr>
                    <ReferenceItemOrigin
                        clonedFrom={clonedFrom}
                        importedFrom={importedFrom}
                        remotesFrom={remotesFrom}
                    />
                    <ReferenceItemBuild id={id} latestBuild={latestBuild} progress={progress} />
                </tbody>
            </ReferenceItemTable>
            <ProgressBar bsStyle={progress === 100 ? "success" : "warning"} now={progress} affixed />
        </StyledReferenceItem>
    );
};

export const mapStateToProps = (state, ownProps) => {
    const {
        cloned_from,
        created_at,
        id,
        imported_from,
        latest_build,
        name,
        organism,
        remotes_from,
        user
    } = state.references.documents[ownProps.index];

    const progress = getReferenceItemProgress(state, ownProps.index);

    return {
        id,
        name,
        organism,
        progress,
        clonedFrom: cloned_from,
        createdAt: created_at,
        importedFrom: imported_from,
        latestBuild: latest_build,
        remotesFrom: remotes_from,
        userId: user.id
    };
};

export default connect(mapStateToProps)(ReferenceItem);
