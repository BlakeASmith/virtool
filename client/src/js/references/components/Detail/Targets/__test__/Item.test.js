import { TargetItem } from "../Item";

describe("<TargetItem />", () => {
    const props = {
        name: "foo",
        description: "bar",
        onEdit: jest.fn(),
        onRemove: jest.fn()
    };

    it("should render", () => {
        const wrapper = shallow(<TargetItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
