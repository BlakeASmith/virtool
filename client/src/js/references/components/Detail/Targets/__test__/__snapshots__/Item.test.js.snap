// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TargetItem /> should render 1`] = `
<Box__BoxGroupSection>
  <Item__NameIcon>
    foo
    <FlexItem
      alignSelf={null}
      basis="auto"
      grow={1}
      shrink={1}
    >
      <Flex
        alignContent="stretch"
        alignItems="center"
        className="pull-right"
        direction="row"
        justifyContent="flex-start"
        wrap="nowrap"
      >
        <Icon
          bsStyle="warning"
          faStyle="fas"
          fixedWidth={false}
          name="edit"
          onClick={[MockFunction]}
          pullRight={false}
          tip="Modify"
        />
        <FlexItem
          alignSelf={null}
          basis="auto"
          grow={0}
          pad={true}
          shrink={1}
        >
          <Icon
            bsStyle="danger"
            faStyle="fas"
            fixedWidth={false}
            name="trash"
            onClick={[MockFunction]}
            pullRight={false}
            tip="Remove"
          />
        </FlexItem>
      </Flex>
    </FlexItem>
  </Item__NameIcon>
  bar
</Box__BoxGroupSection>
`;
